From: Hilko Bengen <bengen@debian.org>
Date: Tue, 5 Jul 2011 18:38:35 +0200
Subject: update gnulib code and test: readlink

test-readlink.h:41: assertion failed
/bin/bash: line 5: 15999 Aborted                 EXEEXT='' srcdir='.' LOCALE_FR='fr_FR' LOCALE_FR_UTF8='fr_FR.UTF-8' LOCALE_JA='ja_JP.EUC-JP' LOCALE_ZH_CN='zh_CN.GB18030' LOCALE_FR_UTF8='fr_FR.UTF-8' LOCALE_FR='fr_FR' LOCALE_FR_UTF8='fr_FR.UTF-8' LOCALE_JA='ja_JP.EUC-JP' LOCALE_ZH_CN='zh_CN.GB18030' MAKE='make' ${dir}$tst
FAIL: test-readlink
---
 daemon/lib/readlink.c        |    2 +-
 daemon/m4/readlink.m4        |    7 ++++---
 daemon/tests/test-readlink.c |    2 +-
 daemon/tests/test-readlink.h |    8 ++++----
 4 files changed, 10 insertions(+), 9 deletions(-)

diff --git a/daemon/lib/readlink.c b/daemon/lib/readlink.c
index 1d9b3d3..35d045b 100644
--- a/daemon/lib/readlink.c
+++ b/daemon/lib/readlink.c
@@ -1,5 +1,5 @@
 /* Stub for readlink().
-   Copyright (C) 2003-2007, 2009-2010 Free Software Foundation, Inc.
+   Copyright (C) 2003-2007, 2009-2011 Free Software Foundation, Inc.
 
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
diff --git a/daemon/m4/readlink.m4 b/daemon/m4/readlink.m4
index 36cd41f..ec737d3 100644
--- a/daemon/m4/readlink.m4
+++ b/daemon/m4/readlink.m4
@@ -1,5 +1,5 @@
-# readlink.m4 serial 9
-dnl Copyright (C) 2003, 2007, 2009, 2010 Free Software Foundation, Inc.
+# readlink.m4 serial 10
+dnl Copyright (C) 2003, 2007, 2009-2011 Free Software Foundation, Inc.
 dnl This file is free software; the Free Software Foundation
 dnl gives unlimited permission to copy and/or distribute it,
 dnl with or without modifications, as long as this notice is preserved.
@@ -48,7 +48,8 @@ AC_DEFUN([gl_FUNC_READLINK],
   fi
 ])
 
-# Like gl_FUNC_READLINK, except prepare for separate compilation (no AC_LIBOBJ).
+# Like gl_FUNC_READLINK, except prepare for separate compilation
+# (no REPLACE_READLINK, no AC_LIBOBJ).
 AC_DEFUN([gl_FUNC_READLINK_SEPARATE],
 [
   AC_CHECK_FUNCS_ONCE([readlink])
diff --git a/daemon/tests/test-readlink.c b/daemon/tests/test-readlink.c
index a144964..3da5fbf 100644
--- a/daemon/tests/test-readlink.c
+++ b/daemon/tests/test-readlink.c
@@ -1,5 +1,5 @@
 /* Tests of readlink.
-   Copyright (C) 2009, 2010 Free Software Foundation, Inc.
+   Copyright (C) 2009-2011 Free Software Foundation, Inc.
 
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
diff --git a/daemon/tests/test-readlink.h b/daemon/tests/test-readlink.h
index 1e482df..7247fc4 100644
--- a/daemon/tests/test-readlink.h
+++ b/daemon/tests/test-readlink.h
@@ -1,5 +1,5 @@
 /* Tests of readlink.
-   Copyright (C) 2009, 2010 Free Software Foundation, Inc.
+   Copyright (C) 2009-2011 Free Software Foundation, Inc.
 
    This program is free software: you can redistribute it and/or modify
    it under the terms of the GNU General Public License as published by
@@ -38,7 +38,7 @@ test_readlink (ssize_t (*func) (char const *, char *, size_t), bool print)
   ASSERT (errno == ENOENT);
   errno = 0;
   ASSERT (func ("", buf, sizeof buf) == -1);
-  ASSERT (errno == ENOENT);
+  ASSERT (errno == ENOENT || errno == EINVAL);
   errno = 0;
   ASSERT (func (".", buf, sizeof buf) == -1);
   ASSERT (errno == EINVAL);
@@ -51,7 +51,7 @@ test_readlink (ssize_t (*func) (char const *, char *, size_t), bool print)
   ASSERT (errno == EINVAL);
   errno = 0;
   ASSERT (func (BASE "file/", buf, sizeof buf) == -1);
-  ASSERT (errno == ENOTDIR);
+  ASSERT (errno == ENOTDIR || errno == EINVAL); /* AIX yields EINVAL */
 
   /* Now test actual symlinks.  */
   if (symlink (BASE "dir", BASE "link"))
@@ -74,7 +74,7 @@ test_readlink (ssize_t (*func) (char const *, char *, size_t), bool print)
   ASSERT (symlink (BASE "file", BASE "link2") == 0);
   errno = 0;
   ASSERT (func (BASE "link2/", buf, sizeof buf) == -1);
-  ASSERT (errno == ENOTDIR);
+  ASSERT (errno == ENOTDIR || errno == EINVAL); /* AIX yields EINVAL */
   ASSERT (unlink (BASE "file") == 0);
   ASSERT (unlink (BASE "link2") == 0);
   {
-- 
